packages/ui5-library-writer/src/options/typescript.ts

```diff
@@ -0,0 +1,80 @@
+import { render } from 'ejs';
+import type { Editor } from 'mem-fs-editor';
+import type { UI5LibInput, UI5LibInputTS } from '../types';
+import { getFilePaths } from '@sap-ux/project-access';
+import { mergeObjects, UI5Config } from '@sap-ux/ui5-config';
+import { join } from 'path';
+import { gte } from 'semver';
+
+/**
+ * Enable typescript for the given input.
+ *
+ * @param libInput Input required to enable the optional typescript features
+ * @param basePath - the base path
+ * @param tmplPath - the template path
+ * @param fs - the memfs editor instance
+ */
+export async function enableTypescript(libInput: UI5LibInput, basePath: string, tmplPath: string, fs: Editor) {
+    const tsTmplDirPath = join(tmplPath, 'optional', 'typescript');
+    const tsTmplFilePaths = await getFilePaths(tsTmplDirPath);
+
+    const tsLibInput: UI5LibInputTS = {
+        ...libInput,
+        tsTypes: getTypePackageFor(libInput.framework, libInput.frameworkVersion),
+        tsTypesVersion: libInput.frameworkVersion
+    };
+
+    tsTmplFilePaths.forEach((tsTmplFilePath) => {
+        const relPath = tsTmplFilePath.replace(tsTmplDirPath, '');
+        const outPath = join(basePath, relPath);
+        // Extend or add
+        if (!fs.exists(outPath)) {
+            const separator = process.platform === 'win32' ? '\\' : '/';
```

Comment:
Same question as before, why checking the OS and why don't we do it in any other writer (or did I miss it there when searching the repo)

Reply:
Same as above